src/Symfony/Component/Console/Helper/TreeNode.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+/**
+ * @implements \IteratorAggregate<TreeNode>
+ *
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class TreeNode implements \Countable, \IteratorAggregate
+{
+    /**
+     * @var array<TreeNode|callable(): \Generator>
+     */
+    private array $children = [];
+
+    public function __construct(
+        private readonly string $value = '',
+        ?self $parent = null,
+        iterable $children = [],
+    ) {
+        if ($parent) {
+            $parent->addChild($this);
+        }
+
+        foreach ($children as $child) {
+            $this->addChild($child);
+        }
+    }
+
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    public function addChild(self|string|callable $node): self
+    {
+        if (\is_string($node)) {
+            $node = new self($node, $this);
+        }
+
+        if (\is_callable($node)) {
+            $this->children[] = $node;
+        } else {
+            $this->children[] = $node;
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return \Traversable<TreeNode>
+     */
+    public function getChildren(): \Traversable
+    {
+        foreach ($this->children as $child) {
+            if (\is_callable($child)) {
+                yield from $child();
```

Comment:
```suggestion
                yield from call_user_func($child);
```

Reply:
why ?

Reply:
`()` supports those cases (at least in versions of PHP that we care about in Symfony 7.3. The `callable` type was not consistent with the support for the dynamic function call syntax in PHP 7 and older)